lib/pbench/test/unit/server/test_user_create.py

```diff
@@ -91,12 +92,39 @@ def test_admin_user_creation(server_config, pytestconfig):
                 TestCreateUser.FIRST_NAME_TEXT,
                 TestCreateUser.LAST_NAME_SWITCH,
                 TestCreateUser.LAST_NAME_TEXT,
-                "--entitlement",
+                TestCreateUser.ROLE_SWITCH,
                 "admin",
             ],
             input=f"{TestCreateUser.PSWD_TEXT}\n",
         )
         assert result.exit_code == 0
+        assert (
+            result.stdout
+            == f"{TestCreateUser.PSWD_PROMPT}\n" + "Admin user test_user registered\n"
+        )
+        assert not result.stderr_bytes
+
+    @staticmethod
+    @responses.activate
+    def test_user_creation_with_default_role(server_config, pytestconfig):
+        runner = CliRunner(mix_stderr=False)
+        result = runner.invoke(
+            user_create.main,
+            args=[
+                TestCreateUser.USER_SWITCH,
+                TestCreateUser.USER_TEXT,
+                TestCreateUser.EMAIL_SWITCH,
+                TestCreateUser.EMAIL_TEXT,
+                TestCreateUser.FIRST_NAME_SWITCH,
+                TestCreateUser.FIRST_NAME_TEXT,
+                TestCreateUser.LAST_NAME_SWITCH,
+                TestCreateUser.LAST_NAME_TEXT,
+                TestCreateUser.ROLE_SWITCH,
+                "Default",
+            ],
+            input=f"{TestCreateUser.PSWD_TEXT}\n",
+        )
+        assert result.exit_code == 0
         assert (
             result.stdout
             == f"{TestCreateUser.PSWD_PROMPT}\n" + "User test_user registered\n"
```

Comment:
Why should it fail? we could just ignore the `role` value and create a user with `None` as role value. Which is what this test case is doing, if we specify anything other than defined values in our enum class (at the moment just `ADMIN`), we would just ignore the parameter but create the user where `role` column is not set

Reply:
If as a user I type
```
$ pbench-create-user --username wscales --password hspwd  --role admn --email wscales@redhat.com --first-name Webb --last-name Scales
```
and the command appears to work, then I'm gonna be pissed when I log in and find that I don't have an admin role because my typo was quietly ignored.

Now, if you make `--role` a "choice", then Click should reject that command with an error, and I will see it, fix my typo, and try again.  But, we should have a unit test which proves that Click does that for us....

In general, silent failures (or silently ignoring explicitly specified options) is a Bad Thing™, and it's a Really Bad Thing™ when users are involved.